peer/network.go

```diff
@@ -87,23 +89,25 @@ type network struct {
 	outstandingRequestHandlers map[uint32]message.ResponseHandler // maps avalanchego requestID => message.ResponseHandler
 	activeAppRequests          *semaphore.Weighted                // controls maximum number of active outbound requests
 	activeCrossChainRequests   *semaphore.Weighted                // controls maximum number of active outbound cross chain requests
-	appSender                  common.AppSender                   // avalanchego AppSender for sending messages
-	codec                      codec.Manager                      // Codec used for parsing messages
-	crossChainCodec            codec.Manager                      // Codec used for parsing cross chain messages
-	appRequestHandler          message.RequestHandler             // maps request type => handler
-	crossChainRequestHandler   message.CrossChainRequestHandler   // maps cross chain request type => handler
-	gossipHandler              message.GossipHandler              // maps gossip type => handler
-	peers                      *peerTracker                       // tracking of peers & bandwidth
-	appStats                   stats.RequestHandlerStats          // Provide request handler metrics
-	crossChainStats            stats.RequestHandlerStats          // Provide cross chain request handler metrics
+	router                     *p2p.Router
+	appSender                  common.AppSender                 // avalanchego AppSender for sending messages
+	codec                      codec.Manager                    // Codec used for parsing messages
+	crossChainCodec            codec.Manager                    // Codec used for parsing cross chain messages
+	appRequestHandler          message.RequestHandler           // maps request type => handler
+	crossChainRequestHandler   message.CrossChainRequestHandler // maps cross chain request type => handler
+	gossipHandler              message.GossipHandler            // maps gossip type => handler
+	peers                      *peerTracker                     // tracking of peers & bandwidth
+	appStats                   stats.RequestHandlerStats        // Provide request handler metrics
+	crossChainStats            stats.RequestHandlerStats        // Provide cross chain request handler metrics
 
 	// Set to true when Shutdown is called, after which all operations on this
 	// struct are no-ops.
 	closed utils.Atomic[bool]
 }
 
-func NewNetwork(appSender common.AppSender, codec codec.Manager, crossChainCodec codec.Manager, self ids.NodeID, maxActiveAppRequests int64, maxActiveCrossChainRequests int64) Network {
+func NewNetwork(router *p2p.Router, appSender common.AppSender, codec codec.Manager, crossChainCodec codec.Manager, self ids.NodeID, maxActiveAppRequests int64, maxActiveCrossChainRequests int64) Network {
```

Comment:
imo - we should initialize `requestIDGen` to `MaxUint32` here and then decrement the counter per requestID in `sendAppRequest`.

This guarantees to avoid any overwriting unless we send 4 billion AppRequest messages.

In the case we _do_ send 4 billion requests, we could encounter unexpected behavior if _both_ message handlers have outstanding requests (with the same requestID + nodeID). This wouldn't panic - but the request would never be delivered to one of the handlers.

Currently in coreth we only use `AppRequest` messages for state syncing afaict. Assuming we will never send any mempool bloom filter requests during state sync, we should be fine regardless of the number of requests we send.

Reply:
A separate alternative would be to explicitly separate the `requestID` space by sending odd requests from the coreth client and even requests from the avalanchego sdk.

Reply:
Discussed offline and it seems cleanest to just use even/odd to avoid the slim possibility of a collision

PR to avalanchego: https://github.com/ava-labs/avalanchego/pull/1975/files